.changeset/curvy-otters-jog.md

```diff
@@ -0,0 +1,56 @@
+---
+'astro': minor
+---
+
+Refactor type for integration hooks to be extensible from an interface on the global `Astro` namespace.
```

Comment:
```suggestion
Adds support for writing custom integration hooks to allow runtime interactions between your integration and other integrations.
```
Something of this style maybe, to more explicitly describe the use case? (Vs the change to the code)

Reply:
Coming back to this again, and thinking that because the change primarily is a refactor, it would be nice to assure people that this will not break existing code. So I think we can get both a use case and an implementation change in here. What do you think? (I updated my suggestion